discord_thread_summarizer_using_openai/discord_utils.bal

```diff
@@ -0,0 +1,55 @@
+import ballerina/time;
+import ballerina/http;
+
+const time:Seconds DISCORD_EPOCH = 1420070400000.00;
+
+http:Client discordClient = check new (DISCORD_API_URL);
+
+function getChannelDetails(int channelId) returns Channel|error {
+    string path = string `/channels/${channelId}`;
+    http:Response response = check discordClient->get(path, headers = {
```

Comment:
You can use client resource method rather than building the path and directly do the databinding
```suggestion
    return check discordClient->/channels/[channelId]({
          "Authorization": botToken
    });
```
Check other places as well

References:
- [Path parameter](https://ballerina.io/learn/by-example/http-client-path-parameter/)
- [Query parameter](https://ballerina.io/learn/by-example/http-client-query-parameter/)
- [Header parameter](https://ballerina.io/learn/by-example/http-client-header-parameter/)

Reply:
Addressed in https://github.com/ballerina-guides/ai-samples/pull/69/commits/1dbf76e067fd08cc0430182aa02b7c98d154f3ac

Reply:
Addressed in https://github.com/ballerina-guides/ai-samples/pull/69/commits/c97a6871cef5ffa8820fc29d01eeb8701c3ff8b1